build/cmake/shared_libs.cmake

```diff
@@ -39,8 +39,8 @@ endif()
 if(NANA_STATIC_STDLIB)
     target_link_libraries(nana
         PUBLIC
-            $<$<CXX_COMPILER_ID:GNU>:-static-libgcc>
-            $<$<CXX_COMPILER_ID:Clang>:-static-libstdc++>
+            $<$<CXX_COMPILER_ID:GNU>:-static-libgcc -static-libstdc++>
+            $<$<CXX_COMPILER_ID:Clang>:-static-libgcc -static-libstdc++>
```

Comment:
Recently I learned the following syntax accomplishes the same thing:
`$<$<CXX_COMPILER_ID:GNU,Clang>:-static-libgcc -static-libstdc++> `

See, https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html

> $<CXX_COMPILER_ID:compiler_ids>
> where compiler_ids is a comma-separated list. 1 if the CMake’s compiler id of the CXX compiler matches any one of the entries in compiler_ids, otherwise 0. See also the CMAKE_<LANG>_COMPILER_ID variable.

Reply:
Nice to see. Our CI travis tested on Ubuntu 16 - xenial with gcc 8. 
Do you have experience running travis with macos or at least with clang?

Reply:
Some. Mostly by reusing the docker containers Conan has already vetted. See, https://github.com/ppetraki/conan-sdl2pp/blob/master/.travis.yml

Those osx containers work. I just haven't gotten to supporting that platform yet. The rest is just boilerplate lifted from other conan example packages e.g. .ci/ and build.py .

Reply:
You might find this useful.
https://github.com/conan-io/conan-docker-tools